### ISSUE TITLE
How can this be implemented in the training of sdxl or flux

### ISSUE THREAD

**Issue by dxhark** [juzhengz/LoRI#3]
How can this be implemented in the training of sdxl or flux

**Comment by juzhengz** [comment]
Thanks for the question! To implement LoRI-D for reducing memory consumption, you can directly set `lora_A.requires_grad = False` for the LoRAs, as shown [here](https://github.com/juzhengz/LoRI/blob/main/src/train_lori.py#L95). For LoRI-S, you’ll need to load the masks along with the corresponding parameters, as done [here](https://github.com/juzhengz/LoRI/blob/main/src/mask_trainers.py#L217). Gradients should be masked accordingly during the backward pass as shown [here](https://github.com/juzhengz/LoRI/blob/main/src/mask_trainers.py#L331).

We’re also interested in seeing how LoRI performs in text-to-image generation such as with SDXL or FLUX. If you have any results or insights, we’d love to see them! Feel free to share your findings or even submit a PR.

**Comment by juzhengz** [comment]
Thanks for the question! To implement LoRI-D for reducing memory consumption, you can directly set `lora_A.requires_grad = False` for the LoRAs, as shown [here](https://github.com/juzhengz/LoRI/blob/main/src/train_lori.py#L95). For LoRI-S, you'll need to load the masks along with the corresponding parameters, as done [here](https://github.com/juzhengz/LoRI/blob/main/src/mask_trainers.py#L217). Gradients should be masked accordingly during the backward pass as shown [here](https://github.com/juzhengz/LoRI/blob/main/src/mask_trainers.py#L331).

We're also interested in seeing how LoRI performs in text-to-image generation such as with SDXL or FLUX. If you have any results or insights, we'd love to see them! Feel free to share your findings or even submit a PR.

**Comment by yitao2020** [comment]
We are also paying attention to this LoRI technology and hope it can be used in comfyui. We want to try to deploy this code locally. What are the requirements for the graphics card configuration?

**Comment by juzhengz** [comment]
Thanks again for your interest! Closing the issue for now.
